airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -14,34 +16,58 @@
 import io.airbyte.config.StandardDiscoverCatalogInput;
 import io.airbyte.config.StandardSyncInput;
 import io.airbyte.config.StandardSyncOutput;
+import io.airbyte.config.persistence.ConfigNotFoundException;
 import io.airbyte.protocol.models.AirbyteCatalog;
 import io.airbyte.protocol.models.ConnectorSpecification;
 import io.airbyte.scheduler.models.IntegrationLauncherConfig;
 import io.airbyte.scheduler.models.JobRunConfig;
+import io.airbyte.validation.json.JsonValidationException;
 import io.airbyte.workers.WorkerUtils;
 import io.airbyte.workers.temporal.check.connection.CheckConnectionWorkflow;
 import io.airbyte.workers.temporal.discover.catalog.DiscoverCatalogWorkflow;
+import io.airbyte.workers.temporal.scheduling.ConnectionUpdaterInput;
+import io.airbyte.workers.temporal.scheduling.ConnectionUpdaterWorkflow;
+import io.airbyte.workers.temporal.scheduling.state.WorkflowState;
 import io.airbyte.workers.temporal.spec.SpecWorkflow;
 import io.airbyte.workers.temporal.sync.SyncWorkflow;
+import io.temporal.api.workflow.v1.WorkflowExecutionInfo;
+import io.temporal.api.workflowservice.v1.ListOpenWorkflowExecutionsRequest;
+import io.temporal.api.workflowservice.v1.ListOpenWorkflowExecutionsResponse;
+import io.temporal.client.BatchRequest;
 import io.temporal.client.WorkflowClient;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import java.io.IOException;
 import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 public class TemporalClient {
 
   private final Path workspaceRoot;
   private final WorkflowClient client;
+  private final WorkflowServiceStubs service;
 
   public static TemporalClient production(final String temporalHost, final Path workspaceRoot) {
-    return new TemporalClient(TemporalUtils.createTemporalClient(temporalHost), workspaceRoot);
+    return new TemporalClient(TemporalUtils.createTemporalClient(temporalHost), workspaceRoot, TemporalUtils.createTemporalService(temporalHost));
```

Comment:
We should probably just pass in a temporal service and have the `TemporalClient` create the client internally based off of the service?

Since the impl is:
```
  public static WorkflowClient createTemporalClient(final String temporalHost) {
    final WorkflowServiceStubs temporalService = createTemporalService(temporalHost);
    return WorkflowClient.newInstance(temporalService);
  }
``` 
we're creating redundant service stubs and inviting different configurations.

Reply:
Done, nice catch!